chainerx_cc/chainerx/routines/manipulation.cc

```diff
@@ -695,4 +695,34 @@ std::vector<Array> Split(const Array& ary, std::vector<int64_t> indices, int8_t
     return out;
 }
 
+Array HStack(const std::vector<Array>& arrays) {
+    if (arrays[0].ndim() <= 1) {
```

Comment:
Have been lazy here with error handling, if all arrays are not of same shape, `Concatenate` will throw.